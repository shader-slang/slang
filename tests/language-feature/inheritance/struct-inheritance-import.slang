// struct-inheritance-import.slang

//TEST(compute):COMPARE_COMPUTE:

// This test confirms that the synthesized code for struct type inheritance
// includes proper linkage decorations so that it can work across modules.

import struct_inheritance_imported;

int test(int val)
{
    Derived d;
    d.a = val;

    return 0x100 + d.a*0x10 + getA(d);
}

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):out,name=outputBuffer
RWStructuredBuffer<int> outputBuffer;

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
    uint tid = dispatchThreadID.x;
    int inVal = tid;
    int outVal = test(inVal);
    outputBuffer[tid] = outVal;
}
