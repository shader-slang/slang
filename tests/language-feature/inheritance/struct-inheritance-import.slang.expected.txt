100
111
122
133
